openapi/gateway.json

```diff
@@ -875,8 +875,8 @@
     },
     "/sp/{type}/{slot}/component/{component}/serial-console/detach": {
       "post": {
-        "summary": "Detach the websocket connection attached to the given SP component's serial",
-        "description": "console, if such a connection exists.",
+        "summary": "Detach the websocket connection attached to the given SP component's",
+        "description": "serial console, if such a connection exists.",
```

Comment:
Hmm, Dropshot should probably consider the first paragraph (or maybe the first sentence?) to be the summary -- line breaks affecting this is a bit weird.

Reply:
First paragraph would match the old way of defining endpoints. We rely on this heavily in Nexus.

Reply:
Looks like the logic might have gotten stepped on here:

https://github.com/oxidecomputer/dropshot/pull/1008/files#diff-e164752baf582e03498739811eea1ef8039fabafcbb1ef954a5c0cb8d479cef2L17-L78

Reply:
> First paragraph would match the old way of defining endpoints. We rely on this heavily in Nexus.

Hmm, well it's currently splitting on the first line break, right? Strange, will look tomorrow.

Reply:
Oh yeah, I see you're right: the existing code also worked that way. We just have been diligent about using it correctly in the external API, largely because if we don't, we see the consequence on the [docs site](https://docs.oxide.computer/api/instance_create) — the summaries are used as the sidebar link titles.

Not sure the basic behavior needs to change. Might be enough to just make it easier to notice you've messed up, like with a warning of some kind.

Reply:
> Oh yeah, I see you're right: the existing code also worked that way. We just have been diligent about using it correctly in the external API, largely because if we don't, we see the consequence on the docs site — the summaries are used as the sidebar link titles.

Ahh -- the first line acts as the sidebar title so it's deliberately kept very short for that reason.

I think it's worth exploring better splitting algorithms for sure.